### PR TITLE
UK Admiralty tide data

### PR DESCRIPTION
**Pull request by DigitalMakerCIC**
Files to use UK Admiralty data

### CODE REVIEW

**Comment by TheKitty** [comment]
Please make these minor fixes so the code will pas the Travis CI lint 1.9.2 checks
image

File: PyPortal_Tides/admiralty_tides.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import time
+import board
+from adafruit_pyportal import PyPortal
+from adafruit_bitmap_font import bitmap_font
+from adafruit_display_text.label import Label
+import json
+from secrets import secrets
```
Comment (caternuson) [inline PyPortal_Tides/admiralty_tides.py]: Here's a boiler plate that other programs have used for the secrets.py import:
```python
# Get wifi details and more from a secrets.py file
try:
    from secrets import secrets
except ImportError:
    print("WiFi secrets are kept in secrets.py, please add them there!")
raise
```

Example:
https://github.com/adafruit/Adafruit_Learning_System_Guides/blob/master/PyPortal_OpenWeather/openweather.py#L14

File: PyPortal_Tides/admiralty_tides.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+# Add all the labels to the display
+for label in HI_LABELS + LO_LABELS + [DATE_LABEL]:
+    pyportal.splash.append(label)
+
+def get_tide_info():
+    """Fetch JSON tide time info and return it."""
+
+    # Get raw JSON data
+    raw_info = pyportal.fetch()
+    raw_info = json.loads(raw_info)
```
Comment (caternuson) [inline PyPortal_Tides/admiralty_tides.py]: Were you not able to work with the return from `pyportal.fetch()` directly for some reason?

File: PyPortal_Tides/admiralty_tides_graphical.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
+import time
+import board
+import displayio
+import json
+from secrets import secrets
```
Comment (caternuson) [inline PyPortal_Tides/admiralty_tides_graphical.py]: Same comment as above.

File: PyPortal_Tides/admiralty_tides_graphical.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    time_marker_bitmap[pixel] = 2
+time_marker = displayio.TileGrid(time_marker_bitmap, pixel_shader=palette, x=-MARK_SIZE, y=-MARK_SIZE)
+pyportal.splash.append(time_marker)
+
+def get_tide_data():
+    """Fetch JSON tide data and return parsed results in a list."""
+
+    # Get raw JSON data
+    raw_data = pyportal.fetch()
+    raw_data = json.loads(raw_data)
```
Comment (caternuson) [inline PyPortal_Tides/admiralty_tides_graphical.py]: Same comment as above.

File: PyPortal_Tides/admiralty_tides.py (hunk 1 above)
Comment (DigitalMakerCIC) [reply PyPortal_Tides/admiralty_tides.py]: changes made

File: PyPortal_Tides/admiralty_tides.py (hunk 2 above)
Comment (DigitalMakerCIC) [reply PyPortal_Tides/admiralty_tides.py]: No it returned an array, that led to an error

File: PyPortal_Tides/admiralty_tides_graphical.py (hunk 3 above)
Comment (DigitalMakerCIC) [reply PyPortal_Tides/admiralty_tides_graphical.py]: changes made

File: PyPortal_Tides/admiralty_tides_graphical.py (hunk 4 above)
Comment (DigitalMakerCIC) [reply PyPortal_Tides/admiralty_tides_graphical.py]: Returned a JSON array that caused an error. This resolved it

**Comment by caternuson** [comment]
Very sorry. My bad. It looks like my boiler plate copied in with bad formatting. It stripped out the spaces in front of `raise` for some reason. That line needs to be indented to line up with the `print`.
Let me try again:
```python
# Get wifi details and more from a secrets.py file
try:
    from secrets import secrets
except ImportError:
    print("WiFi secrets are kept in secrets.py, please add them there!")
    raise
```

The other two complaints are trailing-white space:
image

**Comment by DigitalMakerCIC** [comment]
Think that's fixed it and removed trailing space.
